src/components/forms/Input.tsx

```diff
@@ -128,41 +142,111 @@ export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
-        <LeadingAddonStyled>{leadingAddon.addon}</LeadingAddonStyled>
-      )}
-      <LabelStyled
-        variant={variant || contextVariant}
-        hasLeading={hasLeading}
-        hasTrailing={hasTrailing}
-      >
-        {leadingAddon?.inline && (
-          <LeadingInlineAddonStyled>
-            {leadingAddon.addon}
-          </LeadingInlineAddonStyled>
+    <RootInput>
+      <GroupStyled hasError={hasError}>
+        {leadingAddon && !leadingAddon.inline && (
+          <LeadingAddonStyled>{leadingAddon.addon}</LeadingAddonStyled>
         )}
-        <InputStyled id={name} name={name} {...otherProps} />
-        {trailingAddon?.inline && (
-          <TrailingInlineAddonStyled>
-            {trailingAddon.addon}
-          </TrailingInlineAddonStyled>
+
+        <LabelStyled
+          variant={utilityVariant}
+          hasLeading={hasLeading}
+          hasTrailing={hasTrailing}
+        >
+          {leadingAddon?.inline && (
+            <LeadingInlineAddonStyled className="addon">
+              {leadingAddon.addon}
+            </LeadingInlineAddonStyled>
+          )}
+          <InputStyled
+            id={name}
+            name={name}
+            className={inputClassName}
+            {...otherProps}
+          />
+          {trailingAddon?.inline && (
+            <TrailingInlineAddonStyled className="addon">
+              {trailingAddon.addon}
+            </TrailingInlineAddonStyled>
+          )}
+
+          {loading && (
+            <TrailingInlineAddonStyled
+              style={{ height: utilityVariant === 'default' ? 20 : 10 }}
+            >
+              <FullSpinner
+                height={utilityVariant === 'default' ? 20 : 10}
+                width={utilityVariant === 'default' ? 20 : 10}
+              />
+            </TrailingInlineAddonStyled>
+          )}
+        </LabelStyled>
+
+        {trailingAddon && !trailingAddon.inline && (
+          <TrailingAddonStyled>{trailingAddon.addon}</TrailingAddonStyled>
         )}
-      </LabelStyled>
-      {trailingAddon && !trailingAddon.inline && (
-        <TrailingAddonStyled>{trailingAddon.addon}</TrailingAddonStyled>
-      )}
-    </GroupStyled>
+      </GroupStyled>
+
+      <SubText error={error} help={help} valid={valid} />
+    </RootInput>
+  );
+}
+
+function SubText(props: Pick<InputProps, 'help' | 'error' | 'valid'>) {
+  const { error, help, valid } = props;
+
+  const hasErrorMessage = error !== undefined;
+  const hasValidMessage = typeof valid === 'string' && valid !== undefined;
+  const hasValidBoolean = typeof valid === 'boolean' && valid !== undefined;
+
+  const textToDisplay = hasErrorMessage
+    ? error
+    : hasValidMessage
+    ? valid
+    : help || undefined;
+
+  return (
+    <p
+      style={{
+        color: hasErrorMessage
+          ? 'red'
+          : hasValidMessage || hasValidBoolean
+          ? 'green'
+          : undefined,
+      }}
+    >
+      {textToDisplay}
+    </p>
```

Comment:
This component has bugs and can be simplified (easier to explain voice to voice)